In [1]:
from netCDF4 import Dataset, MFDataset, num2date
import sys
import os

In [2]:
flf = Dataset('./Oxygen_annual_statisticalMean_woa13_all_o00_01.nc')

In [3]:
#lists all available variables in the file. 
flf.variables

OrderedDict([('crs', <class 'netCDF4._netCDF4.Variable'>
              int32 crs()
                  grid_mapping_name: latitude_longitude
                  epsg_code: EPSG:4326
                  longitude_of_prime_meridian: 0.0
                  semi_major_axis: 6378137.0
                  inverse_flattening: 298.25723
              unlimited dimensions: 
              current shape = ()
              filling on, default _FillValue of -2147483647 used),
             ('lat', <class 'netCDF4._netCDF4.Variable'>
              float32 lat(lat)
                  standard_name: latitude
                  long_name: latitude
                  units: degrees_north
                  axis: Y
                  bounds: lat_bnds
              unlimited dimensions: 
              current shape = (180,)
              filling on, default _FillValue of 9.969209968386869e+36 used),
             ('lat_bnds', <class 'netCDF4._netCDF4.Variable'>
              float32 lat_bnds(lat, nbounds)
               

In [4]:
#look more at one variable: the annual mean of O2 in seawater.
flf.variables['o_an']

<class 'netCDF4._netCDF4.Variable'>
float32 o_an(time, depth, lat, lon)
    standard_name: volume_fraction_of_oxygen_in_sea_water
    long_name: Objectively analyzed mean fields for volume_fraction_of_oxygen_in_sea_water at standard depth levels.
    cell_methods: area: mean depth: mean time: mean
    grid_mapping: crs
    units: milliliters_per_liter
    _FillValue: 9.96921e+36
unlimited dimensions: 
current shape = (1, 102, 180, 360)
filling on

#### Notes:
Keep an eye on the shapes. They'll tell you a lot about how the data is structured. For example, the variable "o_an" is the
annual mean of O2 concentration in seawater at set depth levels. The shape is defined in the structure: float32 o_an(time, depth, lat, lon) with the atcual shape being (1, 102, 180, 360). This shape indicates that time = 1 (annual mean), depth = 102 depths, +90/-90 lat, +180/-180 lon.

In [5]:
#declare the lat and lon as separate variables
lat = flf.variables['lat'][:]
lon = flf.variables['lon'][:]
(lat[:5], lon[:5])

(masked_array(data=[-89.5, -88.5, -87.5, -86.5, -85.5],
              mask=False,
        fill_value=1e+20,
             dtype=float32),
 masked_array(data=[-179.5, -178.5, -177.5, -176.5, -175.5],
              mask=False,
        fill_value=1e+20,
             dtype=float32))

In [6]:
#How to find out what the depth options are
allDepths = flf.variables['depth'][:]
allDepths

masked_array(data=[0.00e+00, 5.00e+00, 1.00e+01, 1.50e+01, 2.00e+01,
                   2.50e+01, 3.00e+01, 3.50e+01, 4.00e+01, 4.50e+01,
                   5.00e+01, 5.50e+01, 6.00e+01, 6.50e+01, 7.00e+01,
                   7.50e+01, 8.00e+01, 8.50e+01, 9.00e+01, 9.50e+01,
                   1.00e+02, 1.25e+02, 1.50e+02, 1.75e+02, 2.00e+02,
                   2.25e+02, 2.50e+02, 2.75e+02, 3.00e+02, 3.25e+02,
                   3.50e+02, 3.75e+02, 4.00e+02, 4.25e+02, 4.50e+02,
                   4.75e+02, 5.00e+02, 5.50e+02, 6.00e+02, 6.50e+02,
                   7.00e+02, 7.50e+02, 8.00e+02, 8.50e+02, 9.00e+02,
                   9.50e+02, 1.00e+03, 1.05e+03, 1.10e+03, 1.15e+03,
                   1.20e+03, 1.25e+03, 1.30e+03, 1.35e+03, 1.40e+03,
                   1.45e+03, 1.50e+03, 1.55e+03, 1.60e+03, 1.65e+03,
                   1.70e+03, 1.75e+03, 1.80e+03, 1.85e+03, 1.90e+03,
                   1.95e+03, 2.00e+03, 2.10e+03, 2.20e+03, 2.30e+03,
                   2.40e+03, 2.50e

In [9]:
#Convert the depths from float to integer type
allDepthsInt = allDepths.astype(int)
allDepthsInt

masked_array(data=[   0,    5,   10,   15,   20,   25,   30,   35,   40,
                     45,   50,   55,   60,   65,   70,   75,   80,   85,
                     90,   95,  100,  125,  150,  175,  200,  225,  250,
                    275,  300,  325,  350,  375,  400,  425,  450,  475,
                    500,  550,  600,  650,  700,  750,  800,  850,  900,
                    950, 1000, 1050, 1100, 1150, 1200, 1250, 1300, 1350,
                   1400, 1450, 1500, 1550, 1600, 1650, 1700, 1750, 1800,
                   1850, 1900, 1950, 2000, 2100, 2200, 2300, 2400, 2500,
                   2600, 2700, 2800, 2900, 3000, 3100, 3200, 3300, 3400,
                   3500, 3600, 3700, 3800, 3900, 4000, 4100, 4200, 4300,
                   4400, 4500, 4600, 4700, 4800, 4900, 5000, 5100, 5200,
                   5300, 5400, 5500],
             mask=False,
       fill_value=-9223372036854775808)